src/entity/block_entity/sign.rs

```diff
@@ -179,7 +179,9 @@ impl ecs::System for SignRenderer {
                 y_scale: Y_SCALE,
                 x_scale: X_SCALE,
             };
-            state.build(line, format::Color::Black);
+            // TODO: make sure this is correct!
+            // state.build(line, format::Color::Black);
+            state.build_with_optional_color(line, format::Color::Black);
```

Comment:
The naming seems a bit odd here as well, how about adding an Option to the build function